bin/heudiconv

```diff
@@ -26,6 +26,105 @@ import tarfile
 import dicom as dcm
 import dcmstack as ds
 
+import dicom
+import numpy as np
+from nibabel.nicom import csareader
+
+def get_phase_encode(dcm_file):
+
+    ## note x and y are flipped relative to dicom 
+    ## to get direction in nifti
+    ret_dirs = [("x","x-"),("y","y-"),("z","z-")] 
+
+    dcm_dat = dicom.read_file(dcm_file)
+    ## get the direction cosine
+    dcm_image_dir = dcm_dat.ImageOrientationPatient
+    dcm_image_dir_row = (dcm_image_dir[0],dcm_image_dir[1],dcm_image_dir[2])
+    dcm_image_dir_col = (dcm_image_dir[3],dcm_image_dir[4],dcm_image_dir[5])
+    ## get the phase encode
+    dcm_phase_encode = dcm_dat.InPlanePhaseEncodingDirection
+    if dcm_phase_encode == "ROW":
+        dcm_vec = dcm_image_dir_row
+    else:
+        dcm_vec = dcm_image_dir_col
+        
+        
+    ## we now have the direction of the phase encode vector
+    max_index = np.argmax(np.abs(dcm_vec))
+    
+    try:
+        val = csareader.get_csa_header(dcm_dat)['tags']['PhaseEncodingDirectionPositive']['items'][0]
+    except:
+        return None
+    
+    return ret_dirs[max_index][val]
+
+def get_slice_direction(dcm_file):
+    
+
+    ret_dirs = [("x","x-"),("y","y-"),("z-","z")] 
```

Comment:
Let's move the discussion about the spec away from the implementation. I'm happy to address your concerns here: https://docs.google.com/document/d/1HFUkAEE-pB-angVcYe6pf_-fVf4sCpOHKesUvfb8Grc/edit#